File: 2265-partition-array-according-to-given-pivot/2265-partition-array-according-to-given-pivot.ts
function pivotArray(nums: number[], pivot: number): number[] {
    const n = nums.length;
    let smaller = [], middle = [], bigger = [];
    let i = 0;
    while (i < n) {
        if (nums[i] < pivot) {
            smaller.push(nums[i]);
        }

        else if (nums[i] == pivot) {
            middle.push(nums[i])
        }

        else {
            bigger.push(nums[i]);
        }
        i++;
    }

    return smaller.concat(middle, bigger);
};

File: 2265-partition-array-according-to-given-pivot/2265-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int n = nums.length;
        int[] res = new int[n];

        int i = 0, j = n - 1;
        int left = 0, right = n - 1;

        while (i < n) {
            if (nums[i] < pivot)
                res[left++] = nums[i];
            if (pivot < nums[j])
                res[right--] = nums[j];

            i++;
            j--;
        }

        while (left <= right) {
            res[left++] = pivot;
        }

        return res;
    }
}